hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -755,20 +769,50 @@ private class Emit(
                       SafeRow.getClass, "read",
                       mb.getPType(t.physicalType), region, key.value[Long])
                 }))
-	    val groupRVAs = mb.newField[Array[RegionValueAggregator]]("groupRVAs")
-	    
+            val groupRVAs = mb.newField[Array[RegionValueAggregator]]("groupRVAs")
+
             val seq = emit(args(1), rvas = Some(groupRVAs.load()))
-	    
+
+            EmitTriplet(Code(
+              codeI.setup,
+              codeI.m.mux(
+                Code._empty,
+                Code(
+                  groupRVAs := Code.checkcast[KeyedRegionValueAggregator]((rvas.get)(codeI.value[Int])).invoke[Any, Array[RegionValueAggregator]]("getAggs", wrappedKey),
+                  seq.setup))),
+              const(false),
+              Code._empty)
+
+          case AggElementsLengthCheck() =>
+            val len = emit(args(0))
             EmitTriplet(Code(
               codeI.setup,
               codeI.m.mux(
                 Code._empty,
                 Code(
-		  groupRVAs := Code.checkcast[KeyedRegionValueAggregator]((rvas.get)(codeI.value[Int])).invoke[Any, Array[RegionValueAggregator]]("getAggs", wrappedKey),
-		  seq.setup))),
+                  len.setup,
+                  len.m.mux(
+                    Code._empty,
+                    Code.checkcast[ArrayElementsAggregator]((rvas.get) (codeI.value[Int]))
+                      .invoke[Int, Unit]("checkSizeOrBroadcast", coerce[Int](len.v)))))),
               const(false),
               Code._empty)
 
+          case AggElements() =>
+            val idx = emit(args(0))
+
+              // idx never missing, don't need to check
```

Comment:
argh I'd be a little wary of never calling `idx.m` because it could contain some computation that affects `v`. Maybe we should enforce that it never does, but that's not the case currently.